src/analysis/retail/evoker/preservation/modules/talents/Reversion.tsx

```diff
@@ -0,0 +1,125 @@
+    haste: Haste,
+    combatants: Combatants,
+    hotTracker: HotTrackerPrevoker,
+  };
+  protected haste!: Haste;
+  hotTracker!: HotTrackerPrevoker;
+  combatants!: Combatants;
+  totalExtensionTime: number = 0;
+  reversionCritCount: number = 0;
+  reversionExtensions: Attribution[] = [];
+  additionalHealing: number = 0;
+  additionalAbsorbed: number = 0;
+  healing: number = 0;
+  absorbedHealing: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([TALENTS_EVOKER.REVERSION_TALENT, SPELLS.REVERSION_ECHO]),
+      this.reversionHeal,
+    );
+  }
+
+  get totallHealing() {
+    return this.healing + this.absorbedHealing;
+  }
+
+  get totalAdditionaHealing() {
+    return this.additionalAbsorbed + this.additionalHealing;
+  }
+
+  reversionHeal(event: HealEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (!this.hotTracker.hots[targetId] || !this.hotTracker.hots[targetId][spellId]) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][spellId];
+    if (event.tick) {
+      //if crit add extension
+      if (event.hitType === HIT_TYPES.CRIT) {
+        const extensionLength = this.hotTracker.hotInfo[spellId]?.tickPeriod;
+        debug &&
+          console.log(
+            'Current Haste Value: ' +
+              this.haste.current +
+              '. Calculated extension: ' +
+              extensionLength,
+          );
+        this.totalExtensionTime += extensionLength;
+        this.reversionCritCount += 1;
+        const newReversionCrit = HotTracker.getNewAttribution(
+          REVERSION_EXTENSION_PREFIX + this.reversionCritCount,
+        );
+        this.reversionExtensions.push(newReversionCrit);
+
+        this.hotTracker.addExtension(
+          newReversionCrit,
+          extensionLength,
+          targetId,
+          spellId,
+          event.timestamp,
+        );
+      }
+      //tally additional tick healing
+      if (hot.originalEnd < event.timestamp) {
+        this.additionalHealing += event.amount || 0;
+        this.additionalAbsorbed += event.absorbed || 0;
+      }
+      this.healing += event.amount || 0;
+      this.absorbedHealing += event.absorbed || 0;
+    }
+  }
+
+  statistic() {
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.CORE(5)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={
+          <ul>
+            <li>Extra ticks: {this.reversionCritCount}</li>
+            <li>Total healing from extensions: {formatNumber(this.totalAdditionaHealing)}</li>
+          </ul>
+        }
+      >
+        <TalentSpellText talent={TALENTS_EVOKER.REVERSION_TALENT}>
+          <ItemHealingDone amount={this.totallHealing} />
+          <br />
+          <>
+            {formatDuration(this.totalExtensionTime)} <small>additional duration from crits</small>
```

Comment:
I think that maybe using average additional duration instead of total duration would be better? (or alternatively include both and have one in the tooltip)

Reply:
Updated 
![image](https://user-images.githubusercontent.com/26779541/202473080-9d8b20d5-f365-450a-89a6-a8f6c8742594.png)

Total extension was moved to a line item in the statbox tooltip